### PR TITLE
Add RainLevel property to the World class

### PR DESCRIPTION
**Pull request by niziul**

### CODE REVIEW

**Comment by kagikn** [comment]
I just realized that `GET_RAIN_LEVEL` and `SET_RAIN` use different values, so let's separate them as different properties. The former reads the current rain level and the latter writes a rain level override. If the override value is negative, which is set to -0.01 on the exe loaded, the game calculates the current rain level as usual. Otherwise, the game simply write the override value to the current rain level value.

Well, I should separate the getter and the setter of `World.WindSpeed`, since they also use different value.

File: source/scripting_v3/GTA/World.cs [hunk 1]
```diff
@@ -229,6 +229,42 @@ public static float GravityLevel
...
+			{
+				if (value < 0f)
+				{
+					value = -1;
+				}
+
+				if (value > 1f)
+				{
+					value = 1f;
+				}
```
Comment (kagikn) [inline source/scripting_v3/GTA/World.cs:262]: We could set a new rain level override as is without clamping IMO. The game may set the current rain level to a value more than 1.0 if `Rain` value is more than 1.0 in weather.xml. Try editing that file and see what happens with `GET_RAIN_LEVEL` if you doubt.

Comment (kagikn) [reply source/scripting_v3/GTA/World.cs:262]: Yeah, but the getter and the setter being separate properties.
```suggestion
		public static float RainLevel => Function.Call<float>(Hash.GET_RAIN_LEVEL);	
		public static float RainLevelOverride
		{
			set => Function.Call(Hash.SET_RAIN, value);
		}
```

**Review by kagikn — APPROVED** [review]
In this way, we can add the getter of `RainLevelOverride` if we want to get the rain level override later.

**Comment by niziul** [comment]
> In this way, we can add the getter of `RainLevelOverride` if we want to get the rain level override later.

I see, makes sense!

**Pull request merged**